modules/core/src/lib/attribute/attribute.js

```diff
@@ -354,17 +354,31 @@ export default class Attribute extends DataColumn {
     }
   }
 
+  // check that the first few elements of the attribute are reasonable
+  /* eslint-disable no-fallthrough */
   _checkAttributeArray() {
     const {value} = this;
-    if (value && value.length >= 4) {
-      const valid =
-        Number.isFinite(value[0]) &&
-        Number.isFinite(value[1]) &&
-        Number.isFinite(value[2]) &&
-        Number.isFinite(value[3]);
+    const limit = Math.min(4, this.size);
+    if (value && value.length && value.length >= limit) {
```

Comment:
`value.length >= limit` should cover the `!value.length` case

Reply:
I recall getting errors like cannot access attribute length.  But i'll double check.

Reply:
must have been something else, i'll change the predicate to what it was before with the new limit var